drift/lib/src/runtime/manager/manager.dart

```diff
@@ -123,6 +152,35 @@ class TableManagerState<
     );
   }
 
+  /// Create a copy of this state with a new active dataclass
+  /// This is used internally to mark a manager for having the mapper applied
+  TableManagerState<
+      $Database,
+      $Table,
+      $Dataclass,
+      $FilterComposer,
+      $OrderingComposer,
+      $CreateCompanionCallback,
+      $UpdateCompanionCallback,
+      $MappedDataclass,
+      $NewActiveDataclass> copyWithNewActiveDataclass<$NewActiveDataclass>() {
+    return TableManagerState(
```

Comment:
There would still be a need for 3 generics. 
The parent state still needs to know 3 things:

1. The actual Dataclass of the Table
2. **IF** we run `withReferences`, what we then be returned.
3. What will `.get` return if we ran it now.

We could use a field to be the deciding factor, but it would not reduce any generics.

Reply:
In that case we can keep it as is.